rpcs3/Emu/Cell/Modules/cellCamera.cpp

```diff
@@ -115,6 +116,40 @@ static bool check_dev_num(s32 dev_num)
 	return dev_num == 0;
 }
 
+template <typename F = std::function<bool()>> requires (std::is_convertible_v<std::invoke_result_t<F>, bool>)
+static error_code check_errors(camera_state state, camera_state to_check = camera_state::uninit, F&& func = [](){ return false; })
+{
+	if (state == camera_state::uninit)
+	{
+		return CELL_CAMERA_ERROR_NOT_INIT;
+	}
+
+	if (state >= to_check)
+	{
+		return CELL_OK;
+	}
+
+	if (func())
+	{
+		return CELL_CAMERA_ERROR_PARAM;
```

Comment:
This assumes that the order never changes in any function, which is naive

Reply:
It does not assume anything, its code. It just helps with common situation.